### PR TITLE
Link

### PR DESCRIPTION
**Pull request by mdemare**

### CODE REVIEW

**Review by remvee — CHANGES_REQUESTED** [review]
Do not forget to update CLI.md, doc/design/rio-mapper-ontwerp.org and doc/design/rio-mapper-api.org.

File: src/nl/surf/eduhub_rio_mapper/processing.clj [hunk 1]
```diff
@@ -199,6 +205,107 @@
...
+        (let [rio-obj  (xmlclj->hiccup rio-obj)
+              rio-obj (map #(if (and (vector? %)
+                                     (= :duo:opleidingseenheidcode (first %)))
+                              (assoc % 0 :duo:opleidingseenheidSleutel)
+                              %)
+                           rio-obj)
+              finder   (sleutel-finder sleutelnaam)
+              old-id   (some finder rio-obj)
+              new-id   id
+              rio-new  (mapv (sleutel-changer new-id finder) rio-obj)
```
[the remark this answers is not in the document]
Comment (mdemare) [reply src/nl/surf/eduhub_rio_mapper/processing.clj:300]: I want a hiccup-style vector, and I want to test for vectorness.

File: src/nl/surf/eduhub_rio_mapper/api.clj [hunk 2]
```diff
@@ -99,16 +99,9 @@
...
-                (select-keys [:institution-schac-home
-                              :institution-oin
-                              :trace-context])
-                (assoc :action "link"
-                       ::ooapi/type (types type)
-                       ::ooapi/id id
-                       ::rio/opleidingscode rio-code))}))
+        {{:keys [rio-code]} :params :as request}
+    (assoc (job-route (assoc-in request [:params :action] "link"))
+      ::rio/opleidingscode rio-code))
```
Comment (remvee) [inline src/nl/surf/eduhub_rio_mapper/api.clj:104]: Shouldn't this be the following (and why are there no test failing on this):

```clojure
(assoc-in (job-route (assoc-in request [:params :action] link))
          [:job ::rio/opleidinscode]
          rio-code)
```

Btw: proper indentation makes this more readable.

**Review by remvee — CHANGES_REQUESTED** [review]

**Review by remvee — CHANGES_REQUESTED** [review]
This: https://github.com/jomco/eduhub-rio-mapper/pull/170/files#r1143438881

And please document link in API.md.

**Review by remvee — APPROVED** [review]

**Pull request merged**